docs/en/operations/server_settings/settings.md

```diff
@@ -554,6 +554,36 @@ If the table doesn't exist, ClickHouse will create it. If the structure of the q
 </query_log>
 ```
 
+## query_masking_rules
+
+Regexp-based rules, which will be applied to queries before storing them in server logs,
```

Comment:
From the code it looks like the masking is applied not only to the query but also to the entire text of the log message. This should be documented as well.

Besides the query text, what are other sources of sensitive data that might end up in the logs? Probably we should have a list here.

Reply:
1) Query fragments (for example "condition secret = 1 moved to prewhere")
2) small data fragments can be dumped when input format can't be parsed (for example CSV)
3) filenames (which can contain secrets, for example for Distributed table)
4) urls & connection settings
5) external error messages (smth like 'can't establish connection to DSN xyz://db?user=www&password')
etc.